### DISCUSSION TITLE
predictions are always 0

### DISCUSSION THREAD

**Discussion by qxjl1010** [scikit-multilearn/scikit-multilearn#281] [Q&A] [answered]
I'm pretty new with skmultiLearn, now I use this for 'Chinese' documents multiple label classification.
The training dataset is quite small(like 200 sentences), even I use sentence IN training dataset, I can only got [0,0,0,0,0,0] as the prediction, can I get some help with this?
Thanks!

My code:

# Import BinaryRelevance from skmultilearn
from skmultilearn.problem_transform import BinaryRelevance

from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer
from sklearn.feature_extraction.text import TfidfTransformer
from sklearn.svm import SVC
from scipy import sparse 
import jieba
import codecs
import numpy as np

from warnings import simplefilter
simplefilter(action='ignore', category=FutureWarning)

Q_list = []
L_list = []

# Read Sentence file
with codecs.open('multi-label-Q.txt',encoding='utf-8') as infile:
    for line in infile:
        Q_list.append(line[:-2])
infile.close()

# Read Label file
with open('multi-label-L.txt') as infile:
    for line in infile:
        tmp_l = line[:-1].split(',')
        L_list.append(tmp_l)
infile.close()

L_list = np.array(L_list)

L_Question_list = []

# Preprocess for Chinese sentences
for line in Q_list:
    seg_list = jieba.lcut(line, cut_all=False)
    q_addSpace = ''
    for w in seg_list:
        q_addSpace = q_addSpace + w + ' '
    L_Question_list.append(q_addSpace[:-1])
    

cv = CountVectorizer()
cv_fit=cv.fit_transform(L_Question_list)

transformer = TfidfTransformer()
tfidf = transformer.fit_transform(cv_fit)

M = sparse.lil_matrix((len(L_list),6), dtype=int)
for i,row in enumerate(L_list):
    count = 0
    for col in row:
        M[i, count] = col
        count += 1

# Setup the classifier
clf = BinaryRelevance(classifier=SVC())

# Train
clf.fit(tfidf, M)

# A sentence in train dataset
x_test = '偏头痛多发于什么年龄层？'
# Preprocess for Chinese sentence
seg_list = jieba.lcut(x_test, cut_all=False)
q_addSpace = ''
for w in seg_list:
    q_addSpace = q_addSpace + w + ' '
X_test = [q_addSpace]
cv_fit2=cv.transform(X_test)
tfidf2 = transformer.transform(cv_fit2)

# Predict
pred = clf.predict(tfidf2)
print(pred.todense())

**Comment by Jopepato** [comment, marked as answer]
Hello, I would suggest you to split the dataset in 80% training and 20% test, better than using only one sentence for the test. From that you can try different base classifiers to see if there are any improvements and check the accuracies of your model.
